lib/server-builder.js

```diff
@@ -107,22 +100,17 @@ class ServerBuilder {
   updateDependencyLines (configurationContent, additionalAppDependencies) {
     const dependencyLines = additionalAppDependencies
       .map(function (dependency) {
-        if (/[\s']/.test(dependency)) {
-          throw new Error(`Single quotes and space characters are disallowed in additional dependencies: ${dependency}`);
+        if (/[\s'\\$]/.test(dependency)) {
+          throw new Error(`Single quotes, dollar characters and whitespace characters are disallowed in additional dependencies: ${dependency}`);
         }
         return `implementation '${dependency}'`;
       })
-      .join('\n');
-
-    return configurationContent.split(/\n/)
-      .map(function (line) {
-        if (line.trim() === GRADLE_DEPENDENCIES_PLACEHOLDER) {
-          return dependencyLines;
-        } else {
-          return line;
-        }
-      })
-      .join('\n');
+      .join('$1');
+
+    return configurationContent.replace(
+      new RegExp(`(\\s*^\\s*)${_.escapeRegExp(GRADLE_DEPENDENCIES_PLACEHOLDER)}\\s*$`, 'gm'),
```

Comment:
Cannot `(\\s*^\\s*)` be just `(^\\s*)` ?

Reply:
`(\\s*^\\s*)` will capture new line characters currently used in the file to make sure implementation lines are correctly added. I'll add comment about it.